main.tf

```diff
@@ -15,10 +15,25 @@ locals {
   validate_event_notification = var.enable_event_notification && var.existing_en_instance_crn == null ? tobool("When setting var.enable_event_notification to true, a value must be passed for var.existing_en_instance_crn") : true
   # tflint-ignore: terraform_unused_declarations
   validate_endpoint = var.enable_event_notification && var.endpoint_type == "public" && var.allowed_network == "private-only" ? tobool("It is not allowed to have conflicting var.endpoint_type and var.allowed_network values.") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_region = var.existing_sm_instance_crn == null && var.region == null ? tobool("When existing_sm_instance_crn is null, a value must be passed for var.region") : true
+}
+
+
+
+locals {
+  parsed_existing_sm_instance_crn = var.existing_sm_instance_crn != null ? split(":", var.existing_sm_instance_crn) : []
+  existing_sm_guid                = length(local.parsed_existing_sm_instance_crn) > 0 ? local.parsed_existing_sm_instance_crn[7] : null
+  existing_sm_region              = length(local.parsed_existing_sm_instance_crn) > 0 ? local.parsed_existing_sm_instance_crn[5] : null
+}
+data "ibm_resource_instance" "sm_instance" {
+  count      = var.existing_sm_instance_crn == null ? 0 : 1
+  identifier = var.existing_sm_instance_crn
 }
 
 # Create Secrets Manager Instance
 resource "ibm_resource_instance" "secrets_manager_instance" {
+  count             = var.existing_sm_instance_crn == null ? 1 : 0
```

Comment:
by adding a count in here, do we not need to add a moved block to prevent a breaking change? Similar to what we had to do here: https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/7b9521b30fb8c04753b8fd6d25e92f1f07462f21/solutions/instances/main.tf#L142-L145